src/operations/run_command.ts

```diff
@@ -1,73 +1,60 @@
-
-  /** @deprecated This is an internal option that has undefined behavior for this API */
-  willRetryWrite?: any;
-  /** @deprecated This is an internal option that has undefined behavior for this API */
-  omitReadPreference?: any;
-  /** @deprecated This is an internal option that has undefined behavior for this API */
-  writeConcern?: any;
-  /** @deprecated This is an internal option that has undefined behavior for this API */
-  explain?: any;
-  /** @deprecated This is an internal option that has undefined behavior for this API */
-  readConcern?: any;
-  /** @deprecated This is an internal option that has undefined behavior for this API */
-  collation?: any;
-  /** @deprecated This is an internal option that has undefined behavior for this API */
-  maxTimeMS?: any;
-  /** @deprecated This is an internal option that has undefined behavior for this API */
-  comment?: any;
-  /** @deprecated This is an internal option that has undefined behavior for this API */
-  retryWrites?: any;
-  /** @deprecated This is an internal option that has undefined behavior for this API */
-  dbName?: any;
-  /** @deprecated This is an internal option that has undefined behavior for this API */
-  authdb?: any;
-  /** @deprecated This is an internal option that has undefined behavior for this API */
-  noResponse?: any;
-
-  /** @internal Used for transaction commands */
-  bypassPinningCheck?: boolean;
 } & BSONSerializeOptions;
 
 /** @internal */
-export class RunCommandOperation<T = Document> extends CommandCallbackOperation<T> {
+export class RunCommandOperation<T = Document> extends AbstractOperation<T> {
   override options: RunCommandOptions;
   command: Document;
 
-  constructor(parent: OperationParent | undefined, command: Document, options?: RunCommandOptions) {
-    super(parent, options);
-    this.options = options ?? {};
+  constructor(parent: Db, command: Document, options: RunCommandOptions) {
+    super(options);
+    this.options = options;
     this.command = command;
+    this.ns = parent.s.namespace.withCollection('$cmd');
   }
 
-  override executeCallback(
-    server: Server,
-    session: ClientSession | undefined,
-    callback: Callback<T>
-  ): void {
-    const command = this.command;
-    this.executeCommandCallback(server, session, command, callback);
+  override async execute(server: Server, session: ClientSession | undefined): Promise<T> {
+    this.server = server;
+    return server.commandAsync(this.ns, this.command, {
+      ...this.options,
+      readPreference: this.readPreference,
+      session
+    }) as TODO_NODE_3286;
   }
 }
 
-export class RunAdminCommandOperation<T = Document> extends RunCommandOperation<T> {
-  constructor(parent: OperationParent | undefined, command: Document, options?: RunCommandOptions) {
-    super(parent, command, options);
-    this.ns = new MongoDBNamespace('admin');
+export class RunAdminCommandOperation<T = Document> extends AbstractOperation<T> {
+  command: Document;
```

Comment:
do we not need to override the option type here?

Reply:
No, the inheritance is from `AbstractOperation` not RunCommand, there isn't shared logic and it ended up just being a special case of runCommand where the namespace is hardcoded to admin. I could also change the usages of this class to RunCommandOperation, and pass in admin from the callers, but the support for the internal options `noResponse`, `bypassPinningCheck` on this class and not the other is nice organizationally I think.